ext/standard/var_unserializer.re

```diff
@@ -1201,6 +1201,110 @@ object ":" uiv ":" ["]	{
 	return object_common(UNSERIALIZE_PASSTHRU, elements, has_unserialize);
 }
 
+"E:" uiv ":" ["]	{
+	size_t len, maxlen;
+	char *str;
+	zend_string *enum_name, *case_name;
+	zend_class_entry *ce;
+
+	if (!var_hash) return 0;
+
+	len = parse_uiv(start + 2);
+	maxlen = max - YYCURSOR;
+	if (maxlen < len || len == 0) {
+		*p = start + 2;
+		return 0;
+	}
+
+	str = (char*)YYCURSOR;
+
+	YYCURSOR += len;
+
+	if (*(YYCURSOR) != '"') {
+		*p = YYCURSOR;
+		return 0;
+	}
+	if (*(YYCURSOR+1) != ';') {
+		*p = YYCURSOR+1;
+		return 0;
+	}
+
+	size_t colon_pos = SIZE_MAX;
+	for (size_t i = 0; i < len; ++i) {
+		if (str[i] == ':') {
+			colon_pos = i;
+			break;
+		}
+	}
+	if (colon_pos == SIZE_MAX) {
+		php_error_docref(NULL, E_WARNING, "Invalid enum name '%.*s' (missing colon)", (int) len, str);
+		return 0;
+	}
+
+	enum_name = zend_string_init_fast(str, -(colon_pos - len) - 1);
```

Comment:
I'm really confused, shouldn't the length be just `colon_pos`?

I think your current tests hide the problem because the class name and case name happen to have the same name.

Reply:
Yeah, they have the same length. I noticed this bug a couple of days ago but haven't gotten around fixing it yet.